### PR TITLE
[FLINK-17654] Move Clock classes to flink-core to make them usable outside runtime

### PR DESCRIPTION
**Pull request by aljoscha**
## What is the purpose of the change

We move the existing `Clock` interface and subclasses to `flink-core` because we want to use them in the new `WatermarkGenerator` implementations.

## Brief change log

  - move interfaces
  - small fixups
  - 

## Verifying this change

- this is a simple code rename/move

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/util/clock/Clock.java [hunk 1]
```diff
@@ -20,21 +20,35 @@
 
 /**
  * A clock that gives access to time. This clock returns two flavors of time:
- * 
- * <p><b>Absolute Time:</b> This refers to real world wall clock time, and it typically
+ *
+ * <h3>Absolute Time</h3>
+ *
+ * <p>This refers to real world wall clock time, and it typically
```
Comment (dawidwys) [inline flink-core/src/main/java/org/apache/flink/util/clock/Clock.java:28]:
```suggestion
 * <p>This refers to real world wall clock time, and it is typically
```

File: flink-core/src/main/java/org/apache/flink/util/clock/ManualClock.java [hunk 2]
```diff
@@ -43,7 +44,21 @@ public long relativeTimeNanos() {
 		return currentTime.get();
 	}
 
+	/**
+	 * Advanced the time by the given duration.
```
Comment (dawidwys) [inline flink-core/src/main/java/org/apache/flink/util/clock/ManualClock.java:51]:
```suggestion
	 * Advances the time by the given duration.
```

File: flink-core/src/main/java/org/apache/flink/util/clock/ManualClock.java [hunk 3]
```diff
@@ -43,7 +44,21 @@ public long relativeTimeNanos() {
...
+	 * Time can also move backwards by supplying a negative value.
+	 * This method performs no overflow check.
+	 */
 	public void advanceTime(long duration, TimeUnit timeUnit) {
 		currentTime.addAndGet(timeUnit.toNanos(duration));
 	}
+
+	/**
+	 * Advanced the time by the given duration.
```
Comment (dawidwys) [inline flink-core/src/main/java/org/apache/flink/util/clock/ManualClock.java:60]:
```suggestion
	 * Advances the time by the given duration.
```

File: flink-core/src/main/java/org/apache/flink/util/clock/Clock.java (hunk 1 above)
Comment (aljoscha) [reply flink-core/src/main/java/org/apache/flink/util/clock/Clock.java:28]: thanks!

**Comment by aljoscha** [comment]
Thanks for the review!

**Pull request closed without merging**